test/src/test_utils.h

```diff
@@ -5,33 +5,67 @@
  */
 #ifndef TIMESCALEDB_TEST_UTILS_H
 #define TIMESCALEDB_TEST_UTILS_H
-#include <postgres.h>
 
+#include <postgres.h>
 #include <access/xact.h>
 
-#define AssertInt64Eq(a, b)                                                                        \
+#include "export.h"
+
+static inline const char *
+strip_path(const char *filename)
+{
+	int i = 0, slash = 0;
+
+	while (filename[i] != '\0')
+	{
+		if (filename[i] == '/' || filename[i] == '\\')
+			slash = i;
+		i++;
+	}
+
+	return &filename[slash + 1];
+}
+
+#define TestFailure(fmt, ...)                                                                      \
+	do                                                                                             \
+	{                                                                                              \
+		elog(ERROR,                                                                                \
+			 "TestFailure @ %s %s:%d | " fmt "",                                                   \
+			 strip_path(__FILE__),                                                                 \
+			 __func__,                                                                             \
+			 __LINE__,                                                                             \
+			 ##__VA_ARGS__);                                                                       \
+		pg_unreachable();                                                                          \
+	} while (0)
+
+#define TestAssertInt64Eq(a, b)                                                                    \
 	do                                                                                             \
 	{                                                                                              \
 		int64 a_i = (a);                                                                           \
 		int64 b_i = (b);                                                                           \
 		if (a_i != b_i)                                                                            \
-		{                                                                                          \
-			elog(ERROR, INT64_FORMAT " != " INT64_FORMAT " @ line %d", a_i, b_i, __LINE__);        \
-		}                                                                                          \
+			TestFailure("(%s == %s) [" INT64_FORMAT " == " INT64_FORMAT "]", #a, #b, a_i, b_i);    \
 	} while (0)
 
-#define AssertPtrEq(a, b)                                                                          \
+#define TestAssertPtrEq(a, b)                                                                      \
```

Comment:
I think it is better if we default to this behavior (generating an error rather than aborting) and introduce something like `AssertPtrEqOrCrash` for the case that we really want the abort.

Reply:
I see no reason why we'd want to abort (crash) in tests over printing a useful error. But I guess your comment was just a general observation?

Reply:
Sort of. I am suggesting to not change the name since we already have names without `Test` in the code and that works fine already and we want this behavior as default. If the followup argument is that we might want to keep the old name to have "aborting behaviour", the followup is that we should rather introduce a new function for this than use the old one.

Reply:
These macros are for use in tests and I don't think they should be confused with "regular" assertions (`Assert()`) used throughout the code (even outside tests). The `Test` prefix makes it clear that these are for tests only and are part of our test tools/suite.

Reply:
Then I misread the code: I assumed these were going to replace all assertions.